### ISSUE TITLE
Mac OS Big Sur, Strange native error

### ISSUE THREAD

**Issue by VladimirBelchenkoKG** [lupidan/apple-signin-unity#85]
Hi! My project works fine on Catalina (logins with Apple), but it fails on Big Sur. This is what I get in Console :

Failed to establish connection between notification center endpoint and daemon service with error: Error Domain=NSCocoaErrorDomain Code=4099 UserInfo={NSDebugDescription=<private>}
User is member of local group
User belongs to local, checking passcode
Authorization failed: Error Domain=NSCocoaErrorDomain Code=4099 UserInfo={NSDebugDescription=<private>}

**Comment by lupidan** [comment]
I don't have a a machine with Big Sur to check this out.
Could you specify the Unity version you are using to build?
Are you sure your codesigning is correct?

**Comment by VladimirBelchenkoKG** [comment]
Unity version is 2017.4.35f1
I have a tiny project for testing this bug, there is just a login button. I have a build. This build works on Catalina and doesn't work on Big Sur, so I assume my code signing done correctly. Also I tested signing with `codesign --verify --verbose` and `spctl --assess --verbose`.
Also I tested this case : I created Xcode application with signing button, built it without signing and signed it manually. It works.
Anyway I use this command for singing :
`find [.app] -name "*.meta" -delete`
`codesign --force --deep -s "[My certificate]" --entitlements "[My entitlements]" [.app]`
And here are my entitlements :
```
<key>com.apple.security.cs.disable-executable-page-protection</key>

  <true/>

<key>com.apple.security.cs.allow-jit</key>

  <true/>

<key>com.apple.application-identifier</key>

  <string>[team id].[app id]</string>

<key>com.apple.developer.applesignin</key>

  <array>

    <string>Default</string>
  </array>
<key>com.apple.developer.team-identifier</key>
  <string>[Team id]</string>
<key>com.apple.security.app-sandbox</key>
  <true/>
<key>com.apple.security.device.camera</key>
  <true/>
<key>com.apple.security.files.user-selected.read-only</key>
  <true/>
<key>com.apple.security.network.client</key>
  <true/>
<key>com.apple.security.personal-information.location</key>
  <true/>
<key>keychain-access-groups</key>
  <array>
    <string>[Team id].*</string>
  </array>`

**Comment by VladimirBelchenkoKG** [comment]
Yesterday I was digging in source of the sign-in plugin and found out, that the problem appears at this line :
`_passwordProvider = [[ASAuthorizationPasswordProvider alloc] init];`
Here is the error :
`[com.apple.authkit] Failed to establish connection between notification center endpoint and daemon service with error: Error Domain=NSCocoaErrorDomain Code=4099 UserInfo={NSDebugDescription=<private>}`

**Comment by lupidan** [comment]
Well, that's definately strange. The provider **is** required to make any requests to Apple. I don't understand why it might be failing for you the moment you create the provider.

Are you sure the error comes from that line? Does it disappear if you comment it out? (You can comment it out, make a new `.bundle` using the included Xcode project, and replace it on your unity project)
Is this happening on an Apple Silicon machine? or an Intel machine?
You are codesigning with a development certificate that has your machine Device ID in it, right?

Unfortunately, I don't have Big Sur available to try and get a repo out of it. But I would most definately try to get answers from Apple using their developer forums.

**Comment by VladimirBelchenkoKG** [comment]
After some digging I found, that this problem is definitely not fault of the plugin. Firstly, I can reproduce this error with official Unity plugin, secondly, if I create my own plugin, it fails too (the same line). However, I fixed it by building sample project with Unity 2020! So. I guess, it's related to changes in build process between 2017 and 2020, I'll find it. Thank you for support and quick answering!
